icu4c/source/common/ucurr.cpp

```diff
@@ -2320,41 +2259,38 @@ ucurr_countCurrencies(const char* locale,
         UResourceBundle *cm = ures_getByKey(rb, CURRENCY_MAP, rb, &localStatus);
 
         // Using the id derived from the local, get the currency data
-        UResourceBundle *countryArray = ures_getByKey(rb, id.data(), cm, &localStatus);
+        LocalUResourceBundlePointer countryArray(ures_getByKey(rb, id.data(), cm, &localStatus));
 
         // process each currency to see which one is valid for the given date
         if (U_SUCCESS(localStatus))
         {
-            for (int32_t i=0; i<ures_getSize(countryArray); i++)
+            for (int32_t i=0; i<ures_getSize(countryArray.getAlias()); i++)
             {
                 // get the currency resource
-                UResourceBundle *currencyRes = ures_getByIndex(countryArray, i, nullptr, &localStatus);
+                LocalUResourceBundlePointer currencyRes(ures_getByIndex(countryArray.getAlias(), i, nullptr, &localStatus));
 
                 // get the from date
                 int32_t fromLength = 0;
-                UResourceBundle *fromRes = ures_getByKey(currencyRes, "from", nullptr, &localStatus);
-                const int32_t *fromArray = ures_getIntVector(fromRes, &fromLength, &localStatus);
+                LocalUResourceBundlePointer fromRes(ures_getByKey(currencyRes.getAlias(), "from", nullptr, &localStatus));
+                const int32_t *fromArray = ures_getIntVector(fromRes.getAlias(), &fromLength, &localStatus);
 
                 int64_t currDate64 = (int64_t)((uint64_t)(fromArray[0]) << 32);
                 currDate64 |= ((int64_t)fromArray[1] & (int64_t)INT64_C(0x00000000FFFFFFFF));
                 UDate fromDate = (UDate)currDate64;
 
-                if (ures_getSize(currencyRes)> 2)
+                if (ures_getSize(currencyRes.getAlias())> 2)
                 {
                     int32_t toLength = 0;
-                    UResourceBundle *toRes = ures_getByKey(currencyRes, "to", nullptr, &localStatus);
-                    const int32_t *toArray = ures_getIntVector(toRes, &toLength, &localStatus);
+                    LocalUResourceBundlePointer toRes(ures_getByKey(currencyRes.getAlias(), "to", nullptr, &localStatus));
+                    const int32_t *toArray = ures_getIntVector(toRes.getAlias(), &toLength, &localStatus);
 
                     currDate64 = (int64_t)toArray[0] << 32;
                     currDate64 |= ((int64_t)toArray[1] & (int64_t)INT64_C(0x00000000FFFFFFFF));
                     UDate toDate = (UDate)currDate64;
 
-                    if ((fromDate <= date) && (date < toDate))
-                    {
+                    if ((fromDate <= date) && (date < toDate)) {
```

Comment:
Unrelated formatting change.

Just to be clear about why I think this matters: When I'm debugging ICU and trying to find where and when some behaviour changed and therefore look through the blame log, if it there on a line of code says that the last change was made by a commit with a description like _"Fix error checking …"_ then that's a change I'll have to take a closer look at and if I then see that the change that commit did to this particular line actually wasn't anything but some unrelated formatting change then that's wasted time, but if a change was made by a commit with a description like _"Fix formatting …"_ then I immediately know that I can skip that one, saving time and effort.